File: ThamudicEpiPlatform/csharp/OcrScan.cs
using System.Net.Http;
using System.Net.Http.Headers;
if(args.Length==0){Console.Error.WriteLine("usage: OcrScan image [engine]");return 2;}
using var client=new HttpClient{Timeout=TimeSpan.FromMinutes(2)};
using var form=new MultipartFormDataContent();
using var stream=File.OpenRead(args[0]);
var part=new StreamContent(stream);part.Headers.ContentType=new MediaTypeHeaderValue("application/octet-stream");form.Add(part,"file",Path.GetFileName(args[0]));
var engine=args.Length>1?args[1]:"auto";
var response=await client.PostAsync($"http://127.0.0.1:8010/api/ocr/scan?engine={Uri.EscapeDataString(engine)}",form);
Console.WriteLine(await response.Content.ReadAsStringAsync());
response.EnsureSuccessStatusCode();

File: ThamudicEpiPlatform/go/ocr_scan.go
package main

import("bytes";"fmt";"io";"mime/multipart";"net/http";"os")

func scan(path,engine string) error{f,e:=os.Open(path);if e!=nil{return e};defer f.Close();var b bytes.Buffer;w:=multipart.NewWriter(&b);p,e:=w.CreateFormFile("file",path);if e!=nil{return e};if _,e=io.Copy(p,f);e!=nil{return e};w.Close();r,e:=http.Post("http://127.0.0.1:8010/api/ocr/scan?engine="+engine,w.FormDataContentType(),&b);if e!=nil{return e};defer r.Body.Close();out,_:=io.ReadAll(r.Body);fmt.Println(string(out));if r.StatusCode>=300{return fmt.Errorf("ocr HTTP %d",r.StatusCode)};return nil}
func main(){if len(os.Args)<2{fmt.Println("usage: ocr_scan image [engine]");os.Exit(2)};engine:="auto";if len(os.Args)>2{engine=os.Args[2]};if e:=scan(os.Args[1],engine);e!=nil{panic(e)}}
